lib/utils/patterns.ts

```diff
@@ -217,7 +217,7 @@ export function awaitDevice(uuid: string) {
 			`Waiting for ${deviceName} to come online`,
 		);
 
-		const poll = (): Promise<void> => {
+		const poll = (): any => {
```

Comment:
Required as a result of adding `"@types/bluebird": "3.5.25"` to `package.json`. Well, `any` is to some extent an admission of defeat and the reviewer is challenged to come up with a type expression that works (I've tried several alternatives without success). Note that, before I added `"@types/bluebird": "3.5.25"` to `package.json`, the typescript compiler was simply not checking / not trying to match the `Promise<void>` type against the implementation of the `poll()` function. Once it started checking it, I think it actually revealed that the implementation is type-wise internally inconsistent and would require some refactoring. However, in absence of evidence to the contrary, I am adequately satisfied that the type inconsistency is sufficiently harmless to be brushed aside.

Reply:
I think that the issue is that the sdk is using 3.5.21 and 3.5.25 seems to contain a weird change.
Would 3.5.21 work for you in this?
If you indeed need 3.5.21 for this (let me know to also change that in the next major version of the SDK), you can cast the result of the `isOnline()` method before chaining the `.then`
```
return (balena.models.device.isOnline(uuid) as Promise<boolean>).then(function(isOnline) {
```
IMO this is definitely preferable instead of `Promise<any>`